src/map/map.cpp

```diff
@@ -665,9 +665,9 @@ void display_data::set_unit_arrows(std::vector<std::vector<glm::vec2>> const& ar
 }
 
 GLuint load_dds_texture(simple_fs::directory const& dir, native_string_view file_name) {
+	if(!bool(simple_fs::peek_file(dir, file_name)))
```

Comment:
Even peeking at the file can be time consuming if you are doing it every frame. You need to make sure that on the return from where load_dds_texture is called, the texture is set as already loaded, even if the loading "failed" and we have the null handle. (and then you can go back to just using load without peek)

Reply:
That way we won't get stuck in a loop of calling into the file system every frame looking for the same missing texture over and over again

Reply:
They are only ever called when loading

Reply:
When initially building the map

Reply:
Uh, that sounds impossible, since the map is built as part of the scenario, and obviously those textures need to be loaded when rendering